go-controller/hack/update-codegen.sh

```diff
@@ -63,13 +64,6 @@ for crd in ${crds}; do
     --plural-exceptions="EgressQoS:EgressQoSes" \
     "$@"
 
-  echo "Copying apis for $crd"
```

Comment:
Looks like @npinaeva added this fc615a581c

@npinaeva what was this for?

Reply:
so you are running this script from `go-controller`, it will generate the api output under `go-controller/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/$crd/v1/apis` then you need to copy the generated api to our target directory `go-controller/pkg/crd/v1`. Folder removal is needed to ensure no stale files are left in the `go-controller/pkg/crd/v1`, so it removes the directory we want to copy TO not FROM.

The strange part if why the script says `'github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1/apis': No such file or directory` while in the output you shared we see
```
~/go/src/github.com/ovn-org/ovn-kubernetes/go-controller/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd$ ls
adminpolicybasedroute  egressfirewall  egressip  egressqos  egressservice
```
so at least `github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/` exists. Let's try to figure this out in slack

Reply:
ah no no so `'github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1/apis': No such file or directory happens when I run the script with my default `GOPATH` set 
while in the output you shared we see the folder set => happens when I run the script with empty `GOPATH`.

Reply:
we discussed this offline, I think in the old change the repo was not placed in GOPATH or whatever but definitely IMHO this bit of deleting and re-adding seems really wrong to me and there is actually no need to do all that.

Reply:
Ah I finally realized what the problem is, thanks @tssurya  
So our scripts used to work only if you store the repo under GOPATH/src/..., which was a part of GOPATH development mode, which is now deprecated https://go.dev/wiki/GOPATH#gopath-variable
I tried to make it work in any place, but forgot to set one extra argument to actually make it ignore GOPATH and make output folder predictable. Also deleting this internalfolder probably would be nice. let me try to fix that

Reply:
lmk if this makes sense https://github.com/ovn-org/ovn-kubernetes/pull/4209

Reply:
thanks Nadia! Will take a look next week, 
btw next time on leave comment on the existing PR as a review for an alternative suggestion rather than opening new PR to fix the same issue - because otherwise its just two people wasting time on two PR alternatives.

Reply:
> I am not sure I understand what is wrong with opening a new PR. It is a way to explain what I initially tried to implement with this script and it doesn't interfere with your PR here as they go in different directions (revert vs fix).

actually for me this is the fix :D though it looks like it reverted what you did it I didn't intend to revert; it just happened that that's the fix.

> Worst case: I sacrificed 10 min of my time (yet explained the original intent in a way that can be tested) if we decide to go the revert way, best case: I saved you some time fixing the bug I have introduced if we decide to go the fix way. 

right so I wish you had just left a comment here explaining that because then its collaborating and working towards a fix together versus each one doing their thing on top.

> I am sorry I didn't answer your last message in slack, as I was interrupted in the middle of our conversation and forgot to get back, but since you have explained your setup well before that, I had enough info to reproduce the problem.

again this is fine, like I said I was trying to understand your setup just like you asked about mine so that I can consider that in my fix but I never got that info :) ...

Reply:
So the problem with this change is it doesn't work if the repo is located anywhere else except for `GOPATH/src`. Code generation scripts (run e.g. `deepcopy-gen -h`) have this parameter
```
-output-base string               Output base; defaults to $GOPATH/src/ or ./ if $GOPATH is not set. (default "./")
```
that has a default value based on $GOPATH, which I wanted to fix with the original change to this script to make it independent of env variables.